### ISSUE TITLE
Result of check functions isn't returned

### ISSUE THREAD

**Issue by evilpie** [web-ext-experiments/logins#7]
The rejection promise is never returned https://github.com/web-ext-experiments/logins/blob/master/api.js#L86, so calling the check function is basically useless. (I haven't tested this)

**Comment by aswan** [comment]
You're right, that code used to throw and then it got carelessly rewritten.
The fix is easy but it should include tests, I'll try to get to it at some point.

**Comment by bobsilverberg** [comment]
Note to self: This is already fixed, along with the test, in my local m-c repo.
